### DISCUSSION TITLE
[Question] Liquid manipulation task does not update

### DISCUSSION THREAD

**Discussion by robegi** [isaac-sim/IsaacLab#509] [Q&A]
### Question

Good morning,

I am trying to set up a liquid RL task using orbit. I saw it was on the paper among the other tasks, but as others have pointed out those environments seem to be not available yet.

I am trying to do it myself but I am having some issues:

1) I tried to save a simple liquid sphere (using just a particle system, a particle set, a mesh and a physics material) in a .usd file and load it in orbit using the AssetBaseCfg class. However the sphere just spawns and stays suspended in the air without moving. I think that the liquid is actually being simulated but the GUI is not updating it. From my understanding, this is a known problem that can be solved by deactivating the fabric extension. I tried to set the flags associated with it (like disable_fabric for the zero agent) but the problem persists. I don't know if trying to spawn the liquid not from a .usd but directly from the script will help or not.
 Also, more in genreal, is it possible to apply events (like resetting the state) to the AssetBase object? Or does it only work for RigidObject and Articulation?

2) Is there any known way to spawn a liquid using orbit?

3) Is there any possibility to have a code sample from the pouring task that was seen in the paper? I know that if you had the environment ready you would have published it, but even an early version is ok, just to have a hint on what to do.

Thanks for your time.

Kind regards

**Comment by intee25** [comment]
I have the same issue as well! A simple working example with particle system manipulation will help a lot!

**Comment by robegi** [comment]
I finally managed to make the GUI show the moving liquid. By several attempts, the only viable way I found is by deactivating the gpu accelereted simulation. I am afraid this might worsen the performances but I did not find any other way until now.
The important steps are the following:

1) When executing a script, use the ```--cpu``` and ```--disable_fabric``` flags.

2) Add in a post_init the following lines:
```
physx_interface = acquire_physx_interface()
physx_interface.overwrite_gpu_setting(1)
```
This is needed to allow the particle simulation after the cpu flag is set true.

I also share a small example I used:
[mod_joint_pos_env_cfg.txt](https://github.com/user-attachments/files/15571278/mod_joint_pos_env_cfg.txt)

This is a modified version of the already implemented lift task in which I modified the file joint_pos_env_cfg.py. This is not a clean script, I am just sharing it in case someone wants to see how I implemented the code above and also because I switched to directly spawn the particle system in the code, which allows better control of the spawning configuration (though the properties I used for the fluid are chosen randomly here). It is only a temporary version.

In the command line, the command to execute for a single instance of the task with the zero control agent is:
```
./orbit.sh -p source/standalone/environments/zero_agent.py --task Isaac-Lift-Cube-Franka-v0 --num_envs 1 --cpu --disable_fabric
```
When I wrote the code I was still using an older version of Orbit, but I don't think it will be difficult to update it to Isaac Lab.

Hope this was helpful.

Kind Regards
